Add allcontributors to template

lyashevska opened the pull request:
<!--
Thanks for making a pull request to this project.
We have added this PR template to help you help us.
Make sure to read the contributing guidelines and abide to the code of conduct.
See the comments below, fill the required fields, and check the items.
-->

## Related issues

<!-- We normally work with (i) create issue; (ii) discussion if necessary; (iii) create PR. So, at least one of the following should be true:-->

<!-- Option 1, this closes an existing issue. Fill the number below-->
Closes #26 

<!-- Option 2, this is a small fix that arguably won't need an issue. Uncomment below -->
<!--
-->

## Checklist

<!-- mark true if NA -->
<!-- leave PR as draft until all is checked -->
- [x] I am following the [contributing guidelines](https://github.com/abelsiqueira/COPIERTemplate.jl/blob/main/docs/src/90-contributing.md)
- [x] Tests are passing
- [x] Lint workflow is passing
- [x] Docs were updated and workflow is passing
- [x] [CHANGELOG.md](https://github.com/abelsiqueira/COPIERTemplate.jl/blob/main/CHANGELOG.md) was updated

abelsiqueira commented on template/README.md.jinja:
```diff
@@ -12,7 +12,7 @@
 [![DOI](https://zenodo.org/badge/DOI/FIXME)](https://doi.org/FIXME)
 {% if AddCodeOfConduct %}[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg)](CODE_OF_CONDUCT.md){% endif %}
-
+{% if AddAllcontributors %}[![All Contributors](https://img.shields.io/github/all-contributors/{{ PackageOwner}}/{{ PackageName}}.jl?labelColor=5e1ec7&color=c0ffee&style=flat-square))](.all-contributorsrc){% endif %}
```
```suggestion
{% if AddAllcontributors %}[![All Contributors](https://img.shields.io/github/all-contributors/{{ PackageOwner}}/{{ PackageName}}.jl?labelColor=5e1ec7&color=c0ffee&style=flat-square))](.all-contributorsrc){% endif %}

```

abelsiqueira commented on template/README.md.jinja:
```diff
@@ -21,6 +21,16 @@ If you use {{ PackageName }}.jl in your work, please cite using the reference gi
 
 If you want to make contributions of any kind, please first that a look into our [contributing guide directly on GitHub](docs/src/90-contributing.md) or the [contributing page on the website](https://{{ PackageOwner }}.github.io/{{ PackageName }}.jl/dev/contributing/).
 
+### Contributors
```
```suggestion
{% if AddAllcontributors %}
### Contributors

<!-- markdown-link-check-disable -->
```
You need the conditional, but I also added this extra directive to avoid link issues.

abelsiqueira commented on template/README.md.jinja:
```diff
@@ -21,6 +21,16 @@ If you use {{ PackageName }}.jl in your work, please cite using the reference gi
+<!-- markdownlint-restore -->
+<!-- prettier-ignore-end -->
+
+<!-- ALL-CONTRIBUTORS-LIST:END -->
```
```suggestion
<!-- ALL-CONTRIBUTORS-LIST:END -->
{% endif %}

```

abelsiqueira commented on template/docs/src/index.md.jinja:
```diff
@@ -5,3 +5,14 @@ CurrentModule = {{ PackageName }}
 # {{ PackageName }}
 
 Documentation for [{{ PackageName }}](https://github.com/{{ PackageOwner }}/{{ PackageName }}.jl).
+
+### Contributors
+
+<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
+<!-- prettier-ignore-start -->
+<!-- markdownlint-disable -->
+
+<!-- markdownlint-restore -->
+<!-- prettier-ignore-end -->
+
+<!-- ALL-CONTRIBUTORS-LIST:END -->
```
Same as README

abelsiqueira commented on template/{% if AddAllcontributors %}.all-contributorsrc{% endif %}.jinja:
```diff
@@ -0,0 +1,5 @@
+"projectName": {{ PackageName }},
+"projectOwner": {{ PackageOwner }},
+"files": ["README.md", "docs/src/index.md"]
+}
```
This should be fixed by pre-commit.

lyashevska replied:
I did run the pre-commit this time, indentation still seems not to be fixed. I dont know why.

abelsiqueira reviewed (changes requested):
Thanks! I've made some comments.
Also, you need to update the tests (`test/runtests.jl`) adding an entry `AllContributors` to the dictionary.
And, you also need to install and run pre-commit. It should run automatically when installed.

lyashevska commented:
Thanks for your review. I hope I addressed all comments, although I did some squashing which I hopefully worked as intended. Please review this PR again.

abelsiqueira commented on test/runtests.jl:
```diff
@@ -16,6 +16,7 @@ template_options = Dict(
   "AuthorName" => "Test",
   "AuthorEmail" => "test@me.now",
   "AskAdvancedQuestions" => true,
+  "AddAllContributors" => true,
```
```suggestion
  "AddAllcontributors" => true,
```
I think this is the error

lyashevska replied:
ah typo! Thanks for spotting!

abelsiqueira commented on template/docs/src/index.md.jinja:
```diff
@@ -5,3 +5,18 @@ CurrentModule = {{ PackageName }}
 
 Documentation for [{{ PackageName }}](https://github.com/{{ PackageOwner }}/{{ PackageName }}.jl).
+
+{% if AddAllcontributors %}
+### Contributors
```
```suggestion
{% if AddAllcontributors %}## Contributors
```
Not that I think about it, this is more correct.

abelsiqueira commented on template/README.md.jinja:
```diff
@@ -21,6 +22,21 @@ If you use {{ PackageName }}.jl in your work, please cite using the reference gi
 
 If you want to make contributions of any kind, please first that a look into our [contributing guide directly on GitHub](docs/src/90-contributing.md) or the [contributing page on the website](https://{{ PackageOwner }}.github.io/{{ PackageName }}.jl/dev/contributing/).
 
+{% if AddAllcontributors %}
+### Contributors
```
```suggestion
{% if AddAllcontributors %}### Contributors
```
Similar situation as below

abelsiqueira commented:
I took the liberty of making some changes before finishing the review.

abelsiqueira commented on template/{% if AddAllcontributors %}.all-contributorsrc{% endif %}.jinja:
```diff
@@ -0,0 +1,5 @@
+{
+"projectName": {{ PackageName }},
+"projectOwner": {{ PackageOwner }},
+"files": ["README.md", "docs/src/index.md"]
```
```suggestion
  "projectName": {{ PackageName }},
  "projectOwner": {{ PackageOwner }},
  "files": ["README.md", "docs/src/index.md"]
```

abelsiqueira reviewed (approved):
Thanks @lyashevska, it's looking great.

abelsiqueira commented:
@allcontributors please add @lyashevska for code

lyashevska commented on CHANGELOG.md:
Ah of course, forgot to update CHANGELOG. Thanks for that.
